src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -83,6 +83,27 @@ public static void AddProject(this SlnFile slnFile, string fullProjectPath, ILis
 
                 if (solutionFolders != null)
                 {
+                    if (solutionFolders.Any())
+                    {
+                        // Check for any existing projects if there is a new matching solution folder and add them to their own solution folder
```

Comment:
I think "Project" can also mean "Solution Folder" so this is confusing me. Is it trying to say: If the solution folder to add has the same name of an existing solution folder then add it to the existing folder? I don't think so.